pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -292,12 +285,48 @@ func (s *NodeController) DoCheck(node *api.Node) []api.NodeCondition {
 			s.deletePods(node.Name)
 		}
 	}
-
 	conditions = append(conditions, *newReadyCondition)
 
+	// Check Condition: NodeSchedulable
+	oldSchedulableCondition := s.getCondition(node, api.NodeSchedulable)
+	newSchedulableCondition := s.checkNodeSchedulable(node)
```

Comment:
What does it mean if a node is schedulable but not ready?  A schedulable node should be ready. We are supporting plugin scheduler, so we need to make sure each condition is sematically right.